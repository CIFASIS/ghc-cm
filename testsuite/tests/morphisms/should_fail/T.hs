module T where

class F a where
class F a => A a where
class A a => M a where

class AM a where

class morphism AM -> M where
