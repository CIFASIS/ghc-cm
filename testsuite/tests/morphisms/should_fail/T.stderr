
T.hs:9:1: error:
    • Bad morphism declaration
    • No generic way to obtain an instance for the A a superclass of M a
      In the morphism declaration
        ‘class morphism AM -> M -- Defined at T.hs:9:1’
